links/endpoint.go

```diff
@@ -41,13 +41,14 @@ type EndpointGeneric struct {
 	randName string
 }
 
-func NewEndpointGeneric(node Node, iface string) *EndpointGeneric {
+func NewEndpointGeneric(node Node, iface string, link Link) *EndpointGeneric {
 	return &EndpointGeneric{
 		Node:      node,
 		IfaceName: iface,
 		// random name is generated for the endpoint to avoid name collisions
 		// when it is first deployed in the root namespace
 		randName: genRandomIfName(),
+		Link:     link,
```

Comment:
now that I don't quite use link directly in the endpoint, I think this can even be removed...